packages/storybook-webpack-angular-types-plugin/src/lib/webpack-angular-types-plugin/plugin.ts

```diff
@@ -56,23 +60,118 @@ export class WebpackAngularTypesPlugin {
 				smallTsProject.resolveSourceFileDependencies();
 
 				for (const { path, module } of modulesToProcess) {
-					const { classesInformation, interfacesInformation } = generateTypeInformation(
+					const {
+						classesInformation,
+						interfacesInformation,
+						functionsInformation,
+						constantsInformation,
+					} = generateTypeInformation(
 						path,
 						smallTsProject,
 						this.options.excludeProperties,
 					);
-					for (const ci of classesInformation) {
-						this.addClassCodeDocDependency(ci, module);
+					for (const classInformation of classesInformation) {
+						this.addClassCodeDocDependency(classInformation, module);
 					}
-					for (const ii of interfacesInformation) {
-						this.addInterfaceCodeDocDependency(ii, module);
+					for (const interfaceInformation of interfacesInformation) {
+						this.addInterfaceCodeDocDependency(interfaceInformation, module);
+					}
+
+					const groupedExportsInformation = this.groupExportInformation(
+						functionsInformation,
+						constantsInformation,
+					);
+
+					for (const groupedExportInformation of groupedExportsInformation) {
+						this.addGroupedExportsCodeDocDependency(groupedExportInformation, module);
 					}
 				}
 				this.moduleQueue = [];
 			});
 		});
 	}
 
+	/**
+	 * Functions and constants can be annotated with "@group-docs" to collect them into a
+	 * single ArgsTable. Returns information grouped by the "groupBy" property.
+	 *
+	 * @param functionsInformation list of all functions
+	 * @param constantsInformation list of all constants
+	 */
+	private groupExportInformation(
+		functionsInformation: FunctionInformation[],
+		constantsInformation: ConstantInformation[],
+	): GroupedExportInformation[] {
+		const exportsInformationMap = new Map<string, ExportsInformation>();
+
+		for (const functionInformation of functionsInformation) {
+			const name = functionInformation.name;
+
+			exportsInformationMap.set(name, {
+				functionsInformation: [functionInformation],
+				constantsInformation: [],
+			});
+
+			const groupBys = functionInformation.groupBy;
+
+			for (const groupBy of groupBys) {
+				if (groupBy) {
```

Comment:
Why can a `groupBy`-entry be not defined? Should this not be filtered out when evaluating the functions jsdocs in the first place?

Reply:
Yes, you're right. I added that it can be an array later on, before it was `string | undefined`. Forgot to remove the check here.